providers/implementations/encode_decode/encode_key2text.c

```diff
@@ -767,6 +769,47 @@ static int rsa_to_text(BIO *out, const void *key, int selection)
 
 /* ---------------------------------------------------------------------- */
 
+#ifndef OPENSSL_NO_ML_DSA
+static int ml_dsa_to_text(BIO *out, const void *key, int selection)
+{
+    const char *name;
+
+    if (out == NULL || key == NULL) {
+        ERR_raise(ERR_LIB_PROV, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+    if (ossl_ml_dsa_key_get_pub(key) == NULL) {
+        /* Regardless of the |selection|, there must be a public key */
+        ERR_raise(ERR_LIB_PROV, PROV_R_NOT_A_PUBLIC_KEY);
```

Comment:
Note ML-KEM has a different error - they should be consistent IMHO.
Missing sounds like a better error match than not-a to me.
However I see existing usage - which will need updating - as the evp_pkey_dhkem_test.c has a dependency of using the less suitable error reason. I'd change them all over to PROV_R_MISSING_KEY in a separate PR - but this PR should keep alignment with ML-KEM. 

``` 
ERR_raise_data(ERR_LIB_PROV, PROV_R_MISSING_KEY,
                       "no %s key material available",
                       type_label);
```
https://github.com/openssl/openssl/blob/feature/ml-kem/providers/implementations/encode_decode/encode_key2text.c#L637C1-L639C36